compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -307,6 +307,37 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    /// Give appropriate suggestion when encountering `[("a", 0) ("b", 1)]`, where the
+    /// likely intention is to create an array containing tuples.
+    fn identify_bad_array_def(
```

Comment:
```suggestion
    fn maybe_suggest_bad_array_definition(
```
nit: less verbose method name

Reply:
Done.